packages/application-shell/src/version.js

```diff
@@ -0,0 +1,2 @@
+// NOTE: This string will be replaced in the a `prepare` run script by the `scripts/version.js` file.
+export default '__@APPLICATION_KIT_PACKAGE/VERSION_OF_RELEASE__';
```

Comment:
It should. Maybe I am not getting the question right. We keep it in src and keep this magic string. Then we bundle with it and only prior to publishing remove this string with the version to be published only in dist. That means the matching towards this string in src remains simple. Otherwise we'd have to flob for a semver in it of replace the files content.

Reply:
> Then we bundle with it

That's what I don't understand, the file does not seem to be imported anywhere. How do you plan to use it?

```js
import version from 'app-shell/version';
```
or

```js
import { version } from 'app-shell';
```

Reply:
Ah you got a point. Thanks.

Reply:
I'm still a bit confused here. Does this magic string changing happen before or after bundling?

Reply:
How about this: e7dfb16 or we copy it do dist and keep it CJS.

Reply:
> I'm still a bit confused here. Does this magic string changing happen before or after bundling?

After bundling. You run `yarn release` (nothing changes). This triggers `yarn build` in each package. The magic string ends up in `dist`. Then `prepare` is invoced by lerna before each package is released. Then we move into dist and replace the string.

Reply:
I'm assuming you did the research and that this is the most straightforward way to do this.

That said, it feels really weird to rewrite a string inside a file generated by rollup.

Reply:
Yip. :)